### PR TITLE
Update golang docs

### PR DESCRIPTION
**Pull request by meatballhat**
per PR https://github.com/travis-ci/travis-build/pull/1636

### CODE REVIEW

File: user/languages/go.md [hunk 1]
```diff
@@ -50,19 +50,28 @@ handled by [gimme](https://github.com/travis-ci/gimme).
...
+- 1.11.x
+- master
 ```
 {: data-file=".travis.yml"}
 
 
+## Go Modules
+
+If Go 1.11+ is specified and either `GO111MODULE=on` is set or a `go.mod` file
+is present, then it is assumed that [Go
```
Comment (BanzaiMan) [inline user/languages/go.md]: `go.mod` should exist at the repository root, correct?

Comment (meatballhat) [reply user/languages/go.md]: @BanzaiMan yes, but I didn't feel that explaining the rules here was appropriate given that they aren't _our_ rules

File: user/languages/go.md [hunk 2]
```diff
@@ -72,26 +81,23 @@ go_import_path: example.org/pkg/foo
...
-  ```
-  go get -t -v ./...
-  ```
+The default `install` step of `travis_install_go_dependencies <go-version>
+[gobuild-args]` will behave differently depending on the Go version specified,
+as well as the presence of certain environment variables and file paths.
 
-* if go version is older than `1.2`
+If the Go version and environment are detected to support Go 1.11+ Modules or Go
+1.5+ vendoring, then we skip to Makefile detection. In all other cases, file
```
Comment (BanzaiMan) [inline user/languages/go.md]: Should `Makefile` be stylized?

File: user/languages/go.md [hunk 3]
```diff
@@ -72,26 +81,23 @@ go_import_path: example.org/pkg/foo
...
+paths are checked for use of [godep](https://github.com/tools/godep) (see below).
 
-  ```
-  go get ./...
-  ```
+If a Makefile is present by any of the following names, then no further actions
```
Comment (BanzaiMan) [inline user/languages/go.md]: Ditto

File: user/languages/go.md (hunk 2 above)
Comment (meatballhat) [reply user/languages/go.md]: I'm good with either! I didn't in this case because we check for 4 different known file names.

Comment (BanzaiMan) [reply user/languages/go.md]: Ah, I see. I suppose we can leave it as is.

**Pull request merged**